spring-boot-project/spring-boot-devtools/src/main/java/org/springframework/boot/devtools/tunnel/server/SocketTargetServerConnection.java

```diff
@@ -56,9 +56,10 @@ public SocketTargetServerConnection(PortProvider portProvider) {
 	public ByteChannel open(int socketTimeout) throws IOException {
 		SocketAddress address = new InetSocketAddress(this.portProvider.getPort());
 		logger.trace("Opening tunnel connection to target server on " + address);
-		SocketChannel channel = SocketChannel.open(address);
-		channel.socket().setSoTimeout(socketTimeout);
-		return new TimeoutAwareChannel(channel);
+		try (SocketChannel channel = SocketChannel.open(address)) {
```

Comment:
This will close the `SocketChannel` before it can be used by `TimeoutAwareChannel`. It's eventually closed when the `TimeoutAwareChannel` is closed so this is change isn't needed. I'll address this while merging.